File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/ClinicalEye/LasikRSTModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class LASIKRSTModel
    {
        [Key]
        public int Id { get; set; }
        public int MasterId { get; set; }
        public string PachymetryMicrons { get; set; }
        public string PachymetryNotes { get; set; }
        public string FlapDepthMicrons { get; set; }
        public string FlapDepthNotes { get; set; }
        public string AblationDepthMicrons { get; set; }
        public string AblationDepthNotes { get; set; }
        public string PredictedRSTMicrons { get; set; }
        public string PredictedRSTNotes { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public Boolean IsOD { get; set; }

    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical/eye-examination/shared/LasikRST.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms';
import * as moment from 'moment/moment';

export class LasikRSTModel {
  public Id: number = 0;
  public MasterId: number = 0;
  public PachymetryMicrons: string = null;
  public PachymetryNotes: string = null;
  public FlapDepthMicrons: string = null;
  public FlapDepthNotes: string = null;
  public AblationDepthMicrons: string = null;
  public AblationDepthNotes: string = null;
  public PredictedRSTMicrons: string = null;
  public PredictedRSTNotes: string = null;
  public CreatedBy: number = 0;
  public CreatedOn: Date = new Date();
  public IsOD: boolean;

  constructor() {

  }
  dateValidator(control: FormControl): { [key: string]: boolean } {
    var currDate = moment().format('YYYY-MM-DD HH:mm');
    if (control.value) { // gets empty string for invalid date such as 30th Feb or 31st Nov)
      if ((moment(control.value).diff(currDate) > 0)
        || (moment(currDate).diff(control.value, 'years') > 200)) //can select date upto 200 year past from today.
        return { 'wrongDate': true };
    }
    else
      return { 'wrongDate': true };
  }

}
